data_override/data_override.F90

```diff
@@ -569,120 +134,93 @@ subroutine data_override_unset_domains(unset_Atm, unset_Ocean, &
 
   fail_if_not_set = .true. ; if (present(must_be_set)) fail_if_not_set = must_be_set
 
-  if (.not.module_is_initialized) call mpp_error(FATAL, &
```

Comment:
1. `module_is_initialized` now lives in the r4 and r8 versions of the implementation modules, rather than in the wrapper module.
2. The new `atm_mode`, `ocn_mode`, `lnd_mode`, and `ice_mode` variables now keep track of what has and hasn't been initialized, so `mpp_error` still gets called if the user tries to unset an uninitialized domain.